src/OpenSage.Game.Tests/Logic/Object/Update/UpdateModuleTest.cs

```diff
@@ -2,7 +2,7 @@
 
 namespace OpenSage.Tests.Logic.Object.Update;
 
-public abstract class UpdateModuleTest<TModule, TData> : BehaviorModuleTest<TModule, TData> where TModule : UpdateModule where TData : UpdateModuleData, new()
+public abstract class UpdateModuleTest<TModule, TData> : BehaviorModuleTest<TModule, TData> where TModule : UpdateModule where TData : BehaviorModuleData, new()
```

Comment:
I had to change this because `FireWeaponWhenDamagedBehaviorModuleData` inherits from `UpgradeModuleData`, while `FireWeaponWhenDamagedBehavior` inherits from `UpdateModule`.

Reply:
What a pain lol. I suspect it will be fine and it's probably overconstrained (especially given it's a test class)